applications/ParticleMechanicsApplication/python_scripts/particle_mpm_solver.py

```diff
@@ -3,24 +3,54 @@
 # Importing the Kratos Library
 import KratosMultiphysics
 
+# Check that KratosMultiphysics was imported in the main script
+KratosMultiphysics.CheckForPreviousImport()
+KratosMultiphysics.CheckRegisteredApplications("ParticleMechanicsApplication")
+
 # Import applications and dependencies
 import KratosMultiphysics.SolidMechanicsApplication as KratosSolid
 import KratosMultiphysics.ParticleMechanicsApplication as KratosParticle
 
-# Check that KratosMultiphysics was imported in the main script
-KratosMultiphysics.CheckForPreviousImport()
+# Importing the base class
+from python_solver import PythonSolver
+
+def CreateSolver(model, custom_settings):
+
+    domain_size = custom_settings["domain_size"].GetInt()
 
-def CreateSolver(grid_model_part, initial_material_model_part, material_model_part, custom_settings):
-    return ParticleMPMSolver(grid_model_part, initial_material_model_part, material_model_part, custom_settings)
+    ### In MPM three model parts are needed
+    ## Material model part definition
+    material_model_part_name = custom_settings["model_part_name"].GetString()
```

Comment:
the model-parts should be created in the constructor of the solver

Reply:
I have a problem for this Philipp. Will discuss with you tomorrow.